### PR TITLE
Add widgets for Cards and card groups

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
while lining this up i noted that a few details wrt creating __locator__ as well as the constructor-hierachy inwt.core may demonstrate problematic in future

### CODE REVIEW

File: src/widgetastic_patternfly4/card.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        for index, item in enumerate(elements):
+            result.append((index + 1,))
+        return result
+
+
+class CardGroup( GenericLocatorWidget, View):
+    
+    def __init__(self, parent, locator=None, logger=None, **kwargs):
+        View.__init__(self, parent, logger=logger, **kwargs)
+        self.locator = locator or '//section[@class="pf-c-page__main-section"]/div'
```
Comment (quarckster) [inline src/widgetastic_patternfly4/card.py:45]: Is there a reason to have the locator be absolute?

File: src/widgetastic_patternfly4/card.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+from widgetastic.widget import ParametrizedView, ParametrizedViewRequest, GenericLocatorWidget
+from widgetastic.widget import Widget, View
+from widgetastic.xpath import quote
+from itertools import count
+
+class Card(GenericLocatorWidget):
+
+    def __init__(self, parent, locator=None, logger=None, **kwargs):
+        View.__init__(self, parent, logger=logger, **kwargs)
+        self.locator = locator or ".//article[contains(@class, 'pf-c-card')]"
```
Comment (quarckster) [inline src/widgetastic_patternfly4/card.py:13]: Can we have OUIA locator here:

```python
self.locator = locator or ".//*[@data-component-type='PF4/Card']"
```

File: src/widgetastic_patternfly4/card.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    ROOT = ParametrizedLocator("{@locator}")
+
+
+class CardForCardGroup(ParametrizedView ,Card):
```
Comment (quarckster) [inline src/widgetastic_patternfly4/card.py:17]: Why do you inherit `Card` here?

**Review by quarckster — CHANGES_REQUESTED** [review]

File: src/widgetastic_patternfly4/card.py (hunk 2 above)
Comment (RonnyPfannschmidt) [reply src/widgetastic_patternfly4/card.py:13]: structurally that would be incorrect as ouia is not generally implemented, but we can go with it

File: src/widgetastic_patternfly4/card.py (hunk 3 above)
Comment (RonnyPfannschmidt) [reply src/widgetastic_patternfly4/card.py:17]: cards themselfs are components that can be used outside of collections, but they still have behaviours we miht want to add (like the checked state in genera, or action groups

File: src/widgetastic_patternfly4/card.py [hunk 4]
```diff
@@ -42,7 +42,7 @@ class CardGroup( GenericLocatorWidget, View):
     
     def __init__(self, parent, locator=None, logger=None, **kwargs):
         View.__init__(self, parent, logger=logger, **kwargs)
-        self.locator = locator or '//section[@class="pf-c-page__main-section"]/div'
+        self.locator = locator or 'section[@class="pf-c-page__main-section"]/div'
```
Comment (quarckster) [inline src/widgetastic_patternfly4/card.py:45]: I believe it should be `self.locator = locator or './/section[@class="pf-c-page__main-section"]/div'`

**Review by quarckster — CHANGES_REQUESTED** [review]

File: src/widgetastic_patternfly4/card.py (hunk 4 above)
Comment (RonnyPfannschmidt) [reply src/widgetastic_patternfly4/card.py]: fixed, thanks

File: src/widgetastic_patternfly4/card.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply src/widgetastic_patternfly4/card.py]: fixed

**Review by quarckster — APPROVED** [review]

**Pull request merged**